File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int n = nums.size();
        int L = 0,R=n;
        int mid = (L+R)/2;
        while(L<R){
            if (nums[mid]>=target){
                R = mid;
            } else{
                L = mid+1;
            }
            mid = (R+L)/2;
        }
        if(nums[mid] == target) return mid;
        else return -1;

    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        ndict = dict()
        n = len(nums)
        for i in range(n):
            m = nums[i]
            if m in ndict:
                ndict[m]+= 1
            else:
                ndict[m]=1
        mink = -1
        minindex = 0
        result = [-1 for _ in range(k)]
        freq = [-1 for _ in range(k)]
        print(ndict)
        for m in ndict.keys():
            if ndict[m]>mink:
                result[minindex] = m
                freq[minindex] = ndict[m]
                mink = min(freq)
                for i in range(k):
                    if freq[i]==mink:
                        minindex = i
                        break
        return result
            
        